### ISSUE TITLE
hdrs async open can cause memory leak

### ISSUE THREAD

**Issue by fMeow** [Xuanwo/hdrs#129]
Change `async_open` back to `open` massively reduce memory usage. Invesigating.

**Comment by Xuanwo** [comment]
Maybe related to `blocking::Unblock`?

**Comment by fMeow** [comment]
Agree, though I don't have evidence. Maybe `Unblock` does not correctly free memories of data that passed back to original async runtime?
